mathics/builtin/__init__.py

```diff
@@ -37,6 +38,42 @@
 )
 
 
+def contributing_builtin_var(module, name) -> Optional[type]:
+    """
+    Returns getattr(module, name) if it is a Builtin that
+    should contribute with a definition.
+    Otherwise, returns None.
+    """
+    if name.startswith("_"):
+        return None
+
+    var = getattr(module, name)
+
+    if not (isinstance(var, type) and hasattr(var, "__module__")):
+        return None
+
+    # Skip those builtins defined in another module
+    if var.__module__ != module.__name__:
```

Comment:
[`inspect.ismodule`](https://docs.python.org/3/library/inspect.html#inspect.ismodule) is more idiomatic.